tyr/autoscaler/autoscaler.py

```diff
@@ -0,0 +1,115 @@
+from boto.ec2.autoscale import AutoScaleConnection
+from boto.ec2.autoscale import LaunchConfiguration
+from boto.ec2.autoscale import AutoScalingGroup
+import boto.ec2
+import logging
+
+
+class AutoScaler(object):
+    '''
+    Autoscaler setup class
+    '''
+
+    log = logging.getLogger('Clusters.AutoScaler')
+    log.setLevel(logging.DEBUG)
+    ch = logging.StreamHandler()
+    ch.setLevel(logging.DEBUG)
+    formatter = logging.Formatter(
+        '%(asctime)s [%(name)s] %(levelname)s: %(message)s',
+        datefmt='%H:%M:%S')
+    ch.setFormatter(formatter)
+    log.addHandler(ch)
+
+    def __init__(self, launch_configuration,
+                 autoscaling_group,
+                 node_obj,
+                 desired_capacity=1,
+                 max_size=1,
+                 min_size=1,
+                 default_cooldown=300,
+                 availability_zones=None,
+                 subnet_ids=None,
+                 health_check_grace_period=300):
+
+        self.launch_configuration = launch_configuration
+        self.autoscaling_group = autoscaling_group
+        self.desired_capacity = desired_capacity
+
+        # You can set a list of availability zones explicitly, else it will
+        # just use the one from the node object
+        if availability_zones:
+            self.autoscale_availability_zones = availability_zones
+        else:
+            self.autoscale_availability_zones = None
+
+        # If you set these they must match the availability zones
+        if subnet_ids:
+            self.autoscale_subnets = subnet_ids
+        else:
+            self.autoscale_subnets = None #(get the subnet IDs)
+
+        if not subnet_ids and not availability_zones:
+            raise ValueError("Must specify either availability_zones or subnets.")
+
+        if subnet_ids and availability_zones:
+            self.log.warning("Specified both availability_zones and subnets.")
+
+        self.node_obj = node_obj
+        self.max_size = max_size
+        self.min_size = min_size
+        self.default_cooldown = default_cooldown
+        self.health_check_grace_period = health_check_grace_period
+
+    def establish_autoscale_connection(self):
+        try:
+            self.conn = boto.ec2.autoscale.connect_to_region(self.node_obj.region)
+            self.log.info('Established connection to autoscale')
+        except:
```

Comment:
We normally try to [log exceptions][0] before raising them. (Note - that should probably actually be a `critical` level event).

[0]: https://github.com/hudl/Tyr/blob/master/tyr/servers/server.py#L617